src/caretogether-pwa/src/main.tsx

```diff
@@ -15,35 +15,47 @@ import { AppRoutes } from './AppRoutes';
 import RequestBackdrop from './Shell/RequestBackdrop';
 import { ProgressBackdrop } from './Shell/ProgressBackdrop';
 
+import { PostHogProvider } from 'posthog-js/react';
+
 const root = ReactDOM.createRoot(
   document.getElementById('root') as HTMLElement
 );
 root.render(
   <React.StrictMode>
     <AppInsightsContext.Provider value={aiReactPlugin}>
-      <ThemeProvider theme={theme}>
-        <CssBaseline enableColorScheme />
-        <LocalizationProvider dateAdapter={DateAdapter}>
-          <GlobalErrorBoundary>
-            <RecoilRoot>
-              <Router>
-                <AuthenticationWrapper>
-                  <React.Suspense
-                    fallback={
-                      <ProgressBackdrop opaque>
-                        <p>Initializing...</p>
-                      </ProgressBackdrop>
-                    }
-                  >
-                    <AppRoutes />
-                  </React.Suspense>
-                </AuthenticationWrapper>
-              </Router>
-              <RequestBackdrop />
-            </RecoilRoot>
-          </GlobalErrorBoundary>
-        </LocalizationProvider>
-      </ThemeProvider>
+      <PostHogProvider
+        apiKey={import.meta.env.VITE_APP_PUBLIC_POSTHOG_KEY}
+        options={{
+          api_host: import.meta.env.VITE_APP_PUBLIC_POSTHOG_HOST,
+          session_recording: {
+            maskTextSelector: '*',
+          },
+        }}
```

Comment:
For readability, since we'll be modifying these `options` often (especially this year), can you pull these out into their own `const postHogOptions`, preferably exported in a dedicated "instrumentation" file? The `apiKey` can stay here.

Reply:
done